src/sp/it/pl/plugin/PluginBase.kt

```diff
@@ -1,28 +1,51 @@
 package sp.it.pl.plugin
 
+import mu.KLogging
+import sp.it.pl.main.AppUtil.APP
+import sp.it.pl.main.cv
+import sp.it.pl.util.access.V
+import sp.it.pl.util.access.initSync
+import sp.it.pl.util.conf.IsConfig
 import sp.it.pl.util.dev.Idempotent
+import sp.it.pl.util.file.Util.isValidatedDirectory
+import sp.it.pl.util.functional.ifFalse
+import sp.it.pl.util.functional.seqOf
 
-abstract class PluginBase(name: String): Plugin {
+abstract class PluginBase(override val name: String, isEnabledByDefault: Boolean): Plugin {
 
-    override val name = name
-    private var isActive = false
+    @IsConfig(name = "Enable", info = "Enable/disable this plugin")
+    private val enabled by cv(isEnabledByDefault) { V(it).apply { initSync { runWhenReady { enable(it) } } } }
+    private var isRunning = false
+
+    private fun enable(isToBeRunning: Boolean) {
+        val preInitOk = isToBeRunning && seqOf(location, userLocation)
+                .all { isValidatedDirectory(it) }
+                .ifFalse { APP.messagePane.show("Directory $location or $userLocation can not be used.") }
+        val v = isToBeRunning && preInitOk
+        val action = if (v) "starting" else "stopping"
+
+        logger.info { "Plugin $name $action..." }
+        activate(v)
+    }
 
     @Idempotent
     override fun start() {
-        if (!isActive()) onStart()
-        isActive = true
+        if (!isRunning()) onStart()
+        isRunning = true
     }
 
     @Idempotent
     override fun stop() {
-        if (isActive()) onStop()
-        isActive = false
+        val wasRunning = isRunning
+        isRunning = false
+        if (wasRunning) onStop()
     }
 
     internal abstract fun onStart()
 
     internal abstract fun onStop()
 
-    override fun isActive() = isActive
+    override fun isRunning() = isRunning
```

Comment:
Can not - implementations actually define such field. It could be changed later, but I'm more against than for. Normally I consider 'fields' on interfaces a bad idea as interface defines a contract, i.e., behavior which should be functions, not fields.